### ISSUE TITLE
topics inside the repl

### ISSUE THREAD

**Issue by gdanov** [bhauman/lein-figwheel#354]
I have to work on two different parts of the app simultaneously. This means I have two different pages connected to the same repl and it gets messy.
Being able to specify some topic on both ends of the repl so that I can choose to which page I am talking solves the problem.
I have devised a monkey patch solution where I proxy the repl plug-in on the client side and the FigwheelServer on the server. This way each message sent to the browser is decorated with topic label an the repl plugin in the browser filters out the ones not for it.
While full blown middleware solution might be overkill, having chance to plug-in custom message sender (the way I do it in the browser) on the server side would make it less monkeyish.
The code is not extracted in a lib, but I can paste it for reference.

**Comment by bhauman** [comment]
You know that the intended workflow in this situation is to have two different builds and swtich the cljs repl between the two builds?

**Comment by gdanov** [comment]
yes I do, however it makes no sense as the code and the config is the same

**Comment by bhauman** [comment]
The configuration isn't the same. You will different build-ids and thus the messages will get routed accordingly.  This allows you to take advantage of all the automation that figwheel provides. 

However topics is an interesting "live" solution.  We already have multiplexing based on build-id and that solves a lot of the problems.  But yes a more complete solution could involve focussing on a browser.  And this is possible on the server side without much or any client code.  

Must do some thinksy ...

**Comment by gdanov** [comment]
When there is routing based on the location, the build config is the same. I have the same set of files with the same options and output dir, only the (one and only) entrypoint invokes different fn based on the context. I am getting the feeling I am abusing or misunderstanding something as you keep repeating this...well, the case where I need this is chrome plug-in and that's not very common use case, but imagine I want to work simultaneously on the check-out and the inventory pages of a web shop so I need them opened simultaneously, but route the evals to only one of them (let's say some page tests).

very simple solution would be make the keys for the atom listeners fixed so that one could implement poor man's middleware by wrapping the needed listener in his own code. That's what I do in one direction currently, but in the other direction the listener key is random
